pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -546,3 +549,14 @@ func validateGraph(tasks []PipelineTask) *apis.FieldError {
 	}
 	return nil
 }
+
+func validateMatrix(ctx context.Context, tasks []PipelineTask) (errs *apis.FieldError) {
```

Comment:
Does this require additional validation like the functionality provided [here](https://github.com/tektoncd/pipeline/blob/99b8b196ea753af36befda8c0e0e1eaa9490ae68/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L132)? e.g. validating param types and defaults?

Reply:
Yes, will add those in the next step because it's a lot and want to implement this incrementally (targeting all validation to be included in the next release though)

As noted in the commit and PR message:
> We add validation and testing that the field is available in `alpha` only - we will add further validation in upcoming changes.

Reply:
whoops my bad :)

Reply:
no worries, thanks for the review 😀